database/pk.go

```diff
@@ -0,0 +1,160 @@
+package database
+
+import (
+	"errors"
+	"reflect"
+	"strings"
+	"sync"
+
+	"github.com/konveyor/tackle2-hub/model"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+	"gorm.io/gorm/logger"
+)
+
+// PK singleton pk sequence.
+var PK PkSequence
+
+// PkSequence provides a primary key sequence.
+type PkSequence struct {
+	mutex sync.Mutex
+}
+
+// Load highest key for all models.
+func (r *PkSequence) Load(db *gorm.DB, models []any) (err error) {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	for _, m := range models {
+		mt := reflect.TypeOf(m)
+		if mt.Kind() == reflect.Ptr {
+			mt = mt.Elem()
+		}
+		kind := strings.ToUpper(mt.Name())
```

Comment:
This won't work on any tables that have had their name overridden with a TableName method on the model, but the only instance of that we have is on a join table so it should not be relevant. Just pointing it out in the event that changed in the future for some unforeseen reason.